src/extension.ts

```diff
@@ -383,22 +375,99 @@ async function terraformCommand(command: string, languageServerExec = true): Pro
 }
 
 function enabled(): boolean {
-  return config('terraform').get('languageServer.external', false);
+  return config('terraform').get('languageServer.enable', false);
 }
 
-async function migrateLegacySettings() {
-  if (config('terraform').has('languageServer.enabled')) {
-    try {
-      await config('terraform').update(
-        'languageServer',
-        { enabled: undefined, external: true },
-        vscode.ConfigurationTarget.Global,
+async function migrateLegacySettings(ctx: vscode.ExtensionContext) {
+  // User has asked not to check if settings need to be migrated, so return
+  if (ctx.globalState.get('terraform.disableSettingsMigration', false)) {
+    return;
+  }
+
+  // TODO: Short circuit here so we don't check for every activation forever
```

Comment:
👀 I think this was already implemented above, wasn't it?

Reply:
Oh I see - I misunderstood your intentions. Agreed that it's not worth caching though. So I'd just remove the comment then.